crates/bevy_ui/src/flex/convert.rs

```diff
@@ -32,30 +35,34 @@ pub fn from_val_size(
     }
 }
 
-pub fn from_style(scale_factor: f64, value: &Style) -> taffy::style::Style {
-    taffy::style::Style {
-        display: value.display.into(),
-        position_type: value.position_type.into(),
-        flex_direction: value.flex_direction.into(),
-        flex_wrap: value.flex_wrap.into(),
-        align_items: value.align_items.into(),
-        align_self: value.align_self.into(),
-        align_content: value.align_content.into(),
-        justify_content: value.justify_content.into(),
-        position: from_rect(scale_factor, value.position),
-        margin: from_rect(scale_factor, value.margin),
-        padding: from_rect(scale_factor, value.padding),
-        border: from_rect(scale_factor, value.border),
-        flex_grow: value.flex_grow,
-        flex_shrink: value.flex_shrink,
-        flex_basis: from_val(scale_factor, value.flex_basis),
-        size: from_val_size(scale_factor, value.size),
-        min_size: from_val_size(scale_factor, value.min_size),
-        max_size: from_val_size(scale_factor, value.max_size),
-        aspect_ratio: match value.aspect_ratio {
-            Some(value) => taffy::number::Number::Defined(value),
-            None => taffy::number::Number::Undefined,
-        },
+pub fn from_flex_layout(scale_factor: f64, value: FlexLayoutQueryItem<'_>) -> taffy::style::Style {
+    if let Layout::Flex(flex_layout) = value.layout {
+        taffy::style::Style {
+            display: taffy::style::Display::Flex,
```

Comment:
It looks like we are losing the ability to set `Display::None` to hide elements, which is a regression that we don't want? Or do we want all visibility of the UI to be handled through the VisibilityComponent together with everything else in the engine?

Reply:
We could make it similar to CSS where `Display::None` would "remove" the entire component where as no visibility still allocates space for the component, but it doesn't render anything for it.

Reply:
Agreed, we should keep that behavior in this PR.

Reply:
Wait, have we ever had `Display::None` as a visibility / render thing?

We now have the following on the Bevy side:
```rs
pub enum Layout {
    /// Use the absolute position and size specified.
    Position,
    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm.
    Flex(FlexLayout),
}
```

and this on the Taffy side:
```rs
pub enum Display {
    /// The children will follow the flexbox layout algorithm
    Flex,
    /// The children will not be laid out, and will follow absolute positioning
    None,
}
```

In the rendering of Bevy we have 
```rs
pub struct ExtractedUiNode {
    pub stack_index: usize,
    pub transform: Mat4,
    pub background_color: Color,
    pub rect: Rect,
    pub image: Handle<Image>,
    pub atlas_size: Option<Vec2>,
    pub clip: Option<Rect>,
    pub scale_factor: f32,
}
```
which means that only `ComputedVisibility` has any effect on visibility. 

So adding the ability to hide nodes using any form of `Display::None` would be a feature request and should be considered out-of-scope for this PR?